### PR TITLE
Implemented a __repr__ for cupyx.profiler._time._PerfCaseResult

### PR DESCRIPTION
**Pull request by davidegavio**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,4 @@ htmlcov/
 .pytest_cache/
 cupy/.data/
 cupy/cuda/thrust.h
+.history
```
Comment (emcastillo) [inline .gitignore:27]: Please revert changes to this file.

File: cupyx/profiler/_time.py [hunk 2]
```diff
@@ -26,6 +26,14 @@ def __init__(self, name, ts, devices):
         self._ts = ts
         self._devices = devices
 
+    def __repr__(self) -> str:
+        """ Returns a string representation of the object.
+
+        Returns:
+            str: A string representation of the object.
+        """
+        return f'name: {self.name}, ts: {self._ts}, devices: {self._devices}'
```
Comment (emcastillo) [inline cupyx/profiler/_time.py:35]: Unfortunately this does not give any information of the GPU time/ CPU time etc. Ideally we would like `__repr__` to just call `to_string`

**Comment by kmaehashi** [comment]
Closes https://github.com/cupy/cupy/issues/6319

File: cupyx/profiler/_time.py (hunk 2 above)
Comment (davidegavio) [reply cupyx/profiler/_time.py:35]: The call to `to_str(self, show_gpu=False)` method must be with `show_gpu=False` or `True`?

**Comment by emcastillo** [comment]
Please remove `.gitignore` changes

**Review by emcastillo — APPROVED** [review]

**Pull request merged**